src/dense.jl

```diff
@@ -1,193 +1,207 @@
+
+- tspan    ::Vector{T}  output times
+- points   ::Symbol which points are returned: `:specified` only the
+  ones in tspan or `:all` which includes also the step-points of the solver.
+- stopevent  Stop integration at a zero of this function
+- roottol    TODO
+
+"""
+
+immutable DenseOptions{T<:Number} <: Options{T}
+    tspan    ::Vector{T}
+    tstop    ::T
+    # points   ::Symbol
+    # stopevent::S
+    # roottol  ::T
 end
 
-solve(ode::ExplicitODE,
-      stepper::DenseStepper,
-      options::Options) = Solver(ode,stepper,options)
+@compat function (::Type{DenseOptions{T}}){T}(;
+                                              tstop         = T(Inf),
+                                              tspan::Vector = T[tstop],
+                                              # points::Symbol= :all,
+                                              # stopevent::S  = (t,y)->false,
+                                              # roottol       = eps(T)^T(1//3),
+                                              kargs...)
+    DenseOptions{T}(tspan,tstop)
+end
 
-dense(sol::Solver) = solve(sol.ode, DenseStepper(sol), sol.options)
 
 """
 
-The state of the dense stepper
+A stepper specialized in dense output.  It wraps around another
+`Solver` and stores the subsequent steps generated by `Solver` and
+interpolates the results on request (currently this means at the
+output times stored in `options.tspan`).
+
+"""
+immutable DenseStepper{S<:Solver,O<:DenseOptions} <: AbstractStepper
+    solver::S
+    options::O
+end
 
-- s0, s1: Previous steps, used to produce interpolated output
-- solver_state: The state of the associated solver
-- ytmp: work array
+function solve{T,S<:DenseStepper}(ode::ExplicitODE{T},
+                                  ::Type{S};
+                                  method = RKStepperAdaptive{:rk45},
+                                  options...)
+    sol_orig = Solver(ode,method{T}(; options...))
+    dense_options = DenseOptions{T}(; options...)
+    dense_stepper = S(sol_orig,dense_options)
+    return Solver(ode,dense_stepper)
+end
+
+"""
+
+The state of the dense stepper
 
 """
 type DenseState{St<:AbstractState,T,Y} <: AbstractState{T,Y}
-    s0::Step{T,Y}
-    s1::Step{T,Y}
-    last_tout::T
-    first_step
+    tout_i::Int
+    step_prev::Step{T,Y}
+    step_out::Step{T,Y}
     solver_state::St
-    # used for storing the interpolation result
-    ytmp::Y
-    solver_done
 end
 
+output(ds::DenseState) = output(ds.step_out)
 
-function start{O<:ExplicitODE,S<:DenseStepper}(s::Solver{O,S})
-    # extract the real solver
-    solver = s.stepper.solver
-    t0  = solver.ode.t0
-    y0  = solver.ode.y0
-    dy0 = copy(y0)
-    solver.ode.F!(t0,y0,dy0)
-    step0 = Step(t0,copy(y0),copy(dy0))
-    step1 = Step(t0,copy(y0),copy(dy0))
-    solver_state = start(solver)
-    ytmp = copy(y0)
-    return DenseState(step0, step1, t0-1, true, solver_state, ytmp, false)
+function init{O<:ExplicitODE,S<:DenseStepper}(s::Solver{O,S})
```

Comment:
Needs a different name for same reason as above: `DenseStepper` is not an integrator, thus should not follow this protocol.